### PR TITLE
[DataGrid] Update docs to reflect v5 support

### PR DESCRIPTION
**Pull request by DanailH**
To wrap up #441 

I tried applying the changes mentioned in step 2 from [this comment](https://github.com/mui-org/material-ui-x/pull/855#pullrequestreview-569244937) but there were issues with building the docs that I didn't manage to resolve quickly. Eventually, we would need to spend some more time to make it work and have all the examples in v5.

### CODE REVIEW

File: docs/src/pages/components/data-grid/accessibility/accessibility.md [hunk 1]
```diff
@@ -7,6 +7,4 @@ components: DataGrid, XGrid
 
 <p class="description">The Data Grid has complete accessibility support. For instance, every cell is accessible using the keyboard.</p>
 
-> ⚠️ This feature is only available in v4, head to the [v4 documentation page](https://material-ui.com/components/data-grid/) for more details.
->
-> You can follow [issue #441](https://github.com/mui-org/material-ui-x/issues/441) to be notified once we migrate the data grid from v4 to v5.
+> ⚠️ This feature is now available in v5, head to the [v4 documentation page](https://material-ui.com/components/data-grid/) for more details.
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/accessibility/accessibility.md:10]: What do you think about removing notions of the past and instead mention the future?
```suggestion
> ⚠️ The data grid components are supporting v5 and v4. However, no documentation is available for v5 yet. Check the [documentation of v4](https://material-ui.com/components/data-grid/).
```

x the other cases

Comment (DanailH) [reply docs/src/pages/components/data-grid/accessibility/accessibility.md:10]: I was also thinking of changing the way it is displayed. That looks like a warning bar. Can we have something like an infobox (different color)?

Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/accessibility/accessibility.md:10]: The documentation has no support for different types of messages. So far, we have reproduced a model similar to https://reactjs.org/docs/react-dom-server.html 

It's probably something we could bring cc @mbrookes. I have noticed https://developers.google.com/web/tools/workbox/modules/workbox-window doing such:

Capture d’écran 2021-01-19 à 15 18 36

Capture d’écran 2021-01-19 à 15 18 40

Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/accessibility/accessibility.md:10]: In our case, I think that it should be a warning, the page is useless, it's a placeholder.

Comment (mbrookes) [reply docs/src/pages/components/data-grid/accessibility/accessibility.md]: > It's probably something we could bring

Not sure how we'd distinguish one block-quote from another for the purpose of styling?

**Review by oliviertassinari — APPROVED** [review]

File: docs/src/pages/components/data-grid/accessibility/accessibility.md (hunk 1 above)
Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/accessibility/accessibility.md]: @mbrookes Yeah, it would likely require custom HTML.

**Pull request merged**